compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs

```diff
@@ -249,15 +245,11 @@ runRepl opts mainDar replClient ideState = do
         -- TODO[AH] Deduplicate imports.
         let newImports = imp : imports
         -- TODO[AH] Factor out the module render and typecheck step.
-        liftIO $ writeFileUTF8 (fromNormalizedFilePath $ lineFilePath lineNumber)
-            (renderModule dflags newImports lineNumber [] (WildPat noExt)
+        liftIO $ setBufferModified ideState (lineFilePath lineNumber)
+            $ Just $ T.pack (renderModule dflags newImports lineNumber [] (WildPat noExt)
                 (noLoc $ HsApp noExt
                     (noLoc $ HsVar noExt $ noLoc $ mkRdrUnqual $ mkVarOcc "return")
                     (noLoc $ ExplicitTuple noExt [] Boxed)))
-        -- Useful for debugging, probably best to put it behind a --debug flag
```

Comment:
These are out of date now and trying to ensure code in a comment seems a bit of a lost cause so I’ve removed them for now. Happy to add them back under a debugging flag separately.